src/clients/c++/CMakeLists.txt

```diff
@@ -30,23 +30,47 @@ cmake_minimum_required (VERSION 3.5)
 # librequest.so
 #
 find_package(CURL REQUIRED)
-add_library(
-  request SHARED
-  request.cc request.h
-  request_common.cc request_common.h
-  request_http.cc request_grpc.cc
-  $<TARGET_OBJECTS:model-config-library>
-  $<TARGET_OBJECTS:proto-library>
-  $<TARGET_OBJECTS:grpc-library>
-)
+
+# use different syntax for Windows
+if(WIN32)
+  # using static so that linker does not complain on unresolved external symbol
```

Comment:
What symbol is missing? librequest.so shouldn't have any undefined symbols

Reply:
It complains about Error::Success [here](https://github.com/NVIDIA/tensorrt-inference-server/blob/master/src/clients/c%2B%2B/request.h#L719), it works if I change it to `err` instead of using static class member, but then CI complains with [maybe-uninitialized error](https://gitlab-master.nvidia.com/dl/dgx/tensorrtserver/-/jobs/7009505)

Reply:
Did you try using dllimport as described here: https://stackoverflow.com/a/2479921